File: app/src/main/java/com/prography/prography_androidstudy/RegisterActivity.java
package com.prography.prography_androidstudy;

import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;


public class RegisterActivity extends AppCompatActivity {
    public UserDatabase db;
    private TextView btnRegister;
    private EditText etEmail;
    private EditText etPw;
    private EditText etPwCheck;
    private UserModel userModel;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_register);

        /* findVIewByID */
        btnRegister = findViewById(R.id.register_register_button);
        etEmail = findViewById(R.id.register_email);
        etPw = findViewById(R.id.register_pw);
        etPwCheck = findViewById(R.id.register_pwCheck);

        /* UserDataBase */
        db = UserDatabase.getDatabase(RegisterActivity.this);
        userModel = new UserModel(this);

        btnRegister.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String email_string = etEmail.getText().toString();
                String pw_string = etPw.getText().toString();
                String pwCheck_string = etPwCheck.getText().toString();

                /* pw, pwCheck 유효성 검사 */
                if (pw_string.equals(pwCheck_string)) {
                    /* DB에 email, pw 전달 작업 */
                    if (userModel.signUp(email_string, pw_string)) {
                        /* btnRegister 내용 바꾸기 */
                        Toast.makeText(RegisterActivity.this, "계정 생성에 성공하였습니다.", Toast.LENGTH_SHORT).show();
                        btnRegister.setText("로그인 화면으로 돌아가기");
                        btnRegister.setOnClickListener(new View.OnClickListener() {
                            @Override
                            public void onClick(View v) {
                                finish(); // onDestroy
                            }
                        });
                    } else {
                        Toast.makeText(RegisterActivity.this, "동일한 이메일의 계정이 존재합니다.", Toast.LENGTH_SHORT).show();
                    }

                } else {
                    Toast.makeText(RegisterActivity.this, "비밀번호와 비밀번호 확인이 일치하지 않습니다.", Toast.LENGTH_SHORT).show();
                    etEmail.setText("");
                    etPw.setText("");
                    etPwCheck.setText("");
                }
            }
        });
    }
}

File: app/src/main/java/com/prography/prography_androidstudy/View/RegisterActivity.kt
package com.prography.prography_androidstudy.View

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Toast
import com.prography.prography_androidstudy.Contract.RegisterContract
import com.prography.prography_androidstudy.Presenter.RegisterPresenter
import com.prography.prography_androidstudy.R
import kotlinx.android.synthetic.main.activity_register.*

class RegisterActivity : AppCompatActivity(), RegisterContract.View, View.OnClickListener {
    private var presenter: RegisterContract.Presenter? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_register)

        presenter = RegisterPresenter(this@RegisterActivity, applicationContext, this)
        presenter!!.presenterView()
    }

    override fun setView() {
        register_register_button!!.setOnClickListener(this)
    }

    override fun showToast(title: String) {
        Toast.makeText(this, title, Toast.LENGTH_SHORT).show()
    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.register_register_button -> {
                val email = register_email.text.toString()
                val pw = register_pw.text.toString()
                val pwCheck = register_pwCheck.text.toString()
                presenter!!.signUp(email, pw, pwCheck)
            }
        }
    }
}
